rosidl_typesupport_opensplice_c/resource/msg__type_support_c.cpp.template

```diff
@@ -69,30 +69,40 @@ ROSIDL_GET_TYPE_SUPPORT_FUNCTION(@(pkg), @(subfolder), @(msg))();
 have_not_included_primitive_arrays = True
 have_not_included_string = True
 }@
-@[for field in spec.fields]@
-@[if field.type.is_primitive_type()]@
-@[if field.type.is_array and have_not_included_primitive_arrays]@
+{
+from collections import OrderedDict
+field_types = {}
+for field in spec.fields:
+    if not field.type.is_primitive_type():
+        full_field = field.type.pkg_name + '__msg__' + field.type.type
```

Comment:
`BaseType` does contain the packages name: https://github.com/ros2/rosidl/blob/bf8874c8a2286c19590a85e5090fdc43d77cb51f/rosidl_parser/rosidl_parser/__init__.py#L131